PTGLgraphComputation/src/plcc/Main.java

```diff
@@ -5451,7 +5510,140 @@ private static long calculateSkipNeighborNum(Molecule mol1, Molecule mol2, int m
                     chainSkippedRes += (chainANumberResidues * chainBNumberResidues);
                 }
             }
+        }    
+        // - - - transitive contacts between chains (through ligand contacts) - - -
+        //
+
+        if (Settings.getBoolean("PTGLgraphComputation_transitive_contacts")){
+            // Iteration over all connections of each ligand atom
+            MolContactInfo currentMci = null;
+            Molecule resAOld = null;
+            Molecule resBOld = null;
+            Molecule resA;
+            Molecule resB;
+            Atom atomA;
+            Atom atomB;
+            Integer counterTransitiveContacts = 0;
+
+            for(ArrayList<Atom> currentTransitiveAtomList : ligandToProteinAtomContacts.values()){
+                // Check all keys to evaluate contacts
+
+                for(int a = 0; a < currentTransitiveAtomList.size(); a++){
+                    atomA = currentTransitiveAtomList.get(a);
+                    resA = atomA.molecule;
+                    for (int b = a+1; b < currentTransitiveAtomList.size(); b++){
+                        // Retrieve Residues of Atoms
+                        atomB = currentTransitiveAtomList.get(b);
+                        resB = atomB.molecule;
+
+                        if (resA.equals(resB)){
+                            ;
```

Comment:
If you do not use the body of the if why do you use the if at all? You could just write
`if (!resA.equals(resB)){`
and paste the code from the body of the else here, right?